bigtable-hbase-1.x-parent/bigtable-hbase-tools/src/main/java/com/google/cloud/bigtable/hbase/tools/HBaseSchemaTranslator.java

```diff
@@ -348,23 +363,163 @@ public HBaseSchemaTranslator(SchemaTranslationOptions options) throws IOExceptio
+      // configure the schema translator.
+      for (Map.Entry<String, String> entry : schemaMapping.entrySet()) {
+        // Presence of a ":" indicates that the override is for a column family scoped to a table.
+        if (entry.getKey().contains(COLUMN_FAMILY_SEPARATOR)) {
+          // Split the key on separator. First part is the table and second part is column family.
+          String[] tableAndColumnFamily = entry.getKey().split(COLUMN_FAMILY_SEPARATOR, 2);
+          String table = tableAndColumnFamily[0];
+          String cf = tableAndColumnFamily[1];
+          tableScopedColumnFamilyMappings.putIfAbsent(table, new HashMap<>());
+          // entry<K,V>: key is the source table:cf and value is new cf.
+          tableScopedColumnFamilyMappings.get(tableAndColumnFamily[0]).put(cf, entry.getValue());
+        } else {
+          // Add the table mappings.
+          tableNameMappings.put(entry.getKey(), entry.getValue());
+        }
+      }
+    }
+
+    @Override
+    public ClusterSchemaDefinition transform(final ClusterSchemaDefinition originalSchema)
+        throws IOException, DeserializationException {
+      ClusterSchemaDefinition transformedSchema = new ClusterSchemaDefinition();
+
+      // Parse the mappings
+      parseMappingFile();
+
+      // Apply the transformations.
+      for (TableSchemaDefinition tableSchemaDefinition : originalSchema.tableSchemaDefinitions) {
+        String tableName = tableSchemaDefinition.name;
+        HTableDescriptor tableDescriptor = tableSchemaDefinition.getHbaseTableDescriptor();
+
+        // Override the table name if its present in the mapping file
+        if (tableNameMappings.containsKey(tableName)) {
+          tableName = tableNameMappings.get(tableName);
+        }
+        // Rename the table and copy all the other configs, including the column families.
+        HTableDescriptor newTableDescriptor =
+            new HTableDescriptor(TableName.valueOf(tableName), tableDescriptor);
+
+        // For all the column families that need renaming, replace them.
+        if (tableScopedColumnFamilyMappings.containsKey(tableSchemaDefinition.name)) {
+          Map<String, String> cfMap =
+              tableScopedColumnFamilyMappings.get(tableSchemaDefinition.name);
+          for (Map.Entry<String, String> cfEntry : cfMap.entrySet()) {
+            // Its not possible to rename a columnFamily using the HColumnDescriptor interface
+            // So create a new family with same config and replace it in the HTableDescriptor.
+            newTableDescriptor.addFamily(
+                renameFamily(
+                    cfEntry.getValue(), newTableDescriptor.getFamily(cfEntry.getKey().getBytes())));
+            newTableDescriptor.removeFamily(cfEntry.getKey().getBytes());
+          }
+        } else {
+          // All the column families are copied already, nothing else to do.
+        }
+        transformedSchema.addTableSchemaDefinition(
```

Comment:
However it would be good to stay consistent and add a section comment
`// finalize the transformed shcema `